### PR TITLE
Basic Modal in two separate ways

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-703](https://jira.nypl.org/browse/DSD-703)

## This PR does the following:

- Is an exploration for the `Modal` component and how devs might/should use it.

## How has this been tested?

This is just a POC so we'll close this eventually once we agree on how to move forward. So, ignore the failing GA tests.

## Accessibility concerns or updates

-

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr839-lll4qwnkt7xviitqcfqpnq7735kciohu.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc5a737499db652205bde9

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr839-lll4qwnkt7xviitqcfqpnq7735kciohu.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc5a737499db652205bde9

File: src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -1,34 +1,107 @@
...
+  turpis cursus. Est ante in nibh mauris. Ut faucibus pulvinar
+  elementum integer. Mattis ullamcorper velit sed ullamcorper morbi
+  tincidunt ornare massa. Consectetur purus ut faucibus pulvinar
+  elementum integer enim. Accumsan sit amet nulla facilisi morbi.`;
 
-  componentWillUnmount() {
-    document.body.classList.remove("no-scroll");
-  }
+export function ModalTrigger({
+  bodyContent = testBodyContent,
```
Comment (bigfishdesign13) [inline src/components/Modal/Modal.tsx]: Update to `modalBodyContent`.

File: src/components/Modal/Modal.stories.mdx [hunk 2]
```diff
@@ -36,147 +30,108 @@ import { getCategory } from "../../utils/componentCategories";
...
+This DS `ModalTrigger` component renders a button that you click to open the
+modal.
+
+```jsx
+import { ModalTrigger } from "@nypl/design-system-react-components";
+
+//...
+<ModalTrigger
+  buttonText="Button Text"
+  headingText="Heading text"
```
Comment (bigfishdesign13) [inline src/components/Modal/Modal.stories.mdx]: Update to reflect the new naming convention: `modalHeadingText` and `modalBodyContent`.

File: src/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -1,34 +1,107 @@
...
+  buttonText,
+  closeButtonLabel = "Close",
+  headingText,
```
Comment (bigfishdesign13) [inline src/components/Modal/Modal.tsx]: Update to `modalHeadingText`.

**Review by bigfishdesign13 — APPROVED** [review]
I added some comments about the prop names, but those issues can be addressed in the final.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr839-lll4qwnkt7xviitqcfqpnq7735kciohu.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61dc5a737499db652205bde9

**Comment by EdwinGuzman** [comment]
Closing but using code for final DSD-412 ticket.

**Comment by EdwinGuzman** [comment]
@wluisi @bigfishdesign13  let's talk about this soon and see if it'll work in the upcoming work Bill is doing.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr839-gt9wbfjexcxmm0ivt9qtpeb37soj5xfr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626190762e91387b17150894

**Review by nypl-wluisi — APPROVED** [review]
This looks good to me. I like how you wrapped the Chakra hook `useDisclosure` with `useModal` and handle some the heavy lifting there. It makes the dev experience of using DS modal simpler. There is some potential overlap here w/ what I'm trying to do on `MultiSelect` dialog version, but we'd have to discuss that, it might make sense to deal with that after the initial version of `MultiSelect`.

In regards to this specific work, there is also a prop called `defaultIsOpen` in `useDisclosure`, which might be worth adding here? Not sure the value tho of allowing consuming apps to have a modal "open by default"? But just an idea.

**Comment by EdwinGuzman** [comment]
From meeting:

- Expose the `onClose` function in the `useModal` hook
- allow jsx for the body content
- allow "full" size style only for mobile

**Comment by EdwinGuzman** [comment]
This is ready to be tested now. I added an example and you can create your custom open and close buttons. I'll fix the tugboat issue later.

Screen Shot 2022-04-22 at 5 15 09 PM

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr839-igsfimtu3yrifbrtmguioescildxk2ye.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626304fb2e91387b1726e343

File: src/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -1,34 +1,141 @@
...
+    isOpen,
+    onClose,
+    ...rest
+  }: React.PropsWithChildren<BaseModalProps>) => {
+    const defaultSize = "xl";
+    const fullSize = "full";
+    const [size, setSize] = React.useState<string>(defaultSize);
+    const windowDimensions = useWindowSize();
+    React.useEffect(() => {
+      if (windowDimensions.width <= 320) {
```
Comment (bigfishdesign13) [inline src/components/Modal/Modal.tsx]: Change the trigger width to 600px to align with our breakpoints.

File: src/components/Modal/Modal.tsx [hunk 5]
```diff
@@ -1,34 +1,141 @@
...
+        setSize(defaultSize);
+      }
+    }, [windowDimensions.width]);
 
     return (
-      <div
-        // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
-        tabIndex={0}
-        className={`modal ${className}`}
+      <ChakraModal
```
Comment (bigfishdesign13) [inline src/components/Modal/Modal.tsx:60]: This functionality needs to be implemented:
https://v0.chakra-ui.com/modal#modal-overflow-behavior

To do this, add the following prop to the modal component:
```
scrollBehavior=“inside"
```

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
A few small changes to get the component more inline with where we are headed with this component.

**Comment by EdwinGuzman** [comment]
Updated.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr839-igsfimtu3yrifbrtmguioescildxk2ye.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/626304fb2e91387b1726e343

**Review by bigfishdesign13 — APPROVED** [review]
This looks very good now.  We’ll need to adjust some more styles eventually, but the scrolling functionality is now in line with the final component.

**Review by oliviawongnyc — APPROVED** [review]
Looks good! Excited to try out the hook.

**Pull request merged**